### PR TITLE
Cmake 3.5.2

### PR DESCRIPTION
**Pull request by JPeterMugaas**
Attempt to make a better patch

### CODE REVIEW

**Comment by Alexpux** [comment]
Seems you forgot to commit cmake-3.5.2-msys.patch

**Comment by JPeterMugaas** [comment]
Sorry about the omission.  I hope it shows up now.  For some reason, msys2 environment kept insisting on coming into that patch I did not do a thing.

**Comment by fracting** [comment]
@JPeterMugaas if you don't mind to fetch upstream code and rebase on top of latest git HEAD, it will now trigger both 32bit CI build and 64bit CI build.

See https://github.com/Alexpux/MSYS2-packages/commit/dc2e12131d50f2fd98df4c5b9ad8c675b77c8204

You can also activate your own repo in https://tea-ci.org/JPeterMugaas/MSYS2-packages and test on CI server before creating a pull request, this might help the maintainer save some time before reviewing broken patch ;)

See https://sourceforge.net/p/msys2/mailman/message/35107680/ for more information.

**Comment by Alexpux** [comment]
I don't see patch still.

**Comment by Alexpux** [comment]
See build failure https://ci.appveyor.com/project/Alexpux/msys2-packages/build/1.0.378

**Comment by fracting** [comment]
@JPeterMugaas did you forgot to `git mv cmake-3.2.3-msys.patch cmake-3.5.2-msys.patch`, or did you forgot to `git add cmake-3.5.2-msys.patch`?

File: cmake/cmake-3.5.2-msys.patch [hunk 1]
```diff
@@ -26,27 +26,27 @@ diff -Naur cmake-2.8.11.2/bootstrap cmake-2.8.11.2-msys/bootstrap
...
---- cmake-2.8.11.2/CMakeCPack.cmake	2013-07-02 17:41:39.000000000 +0400
-+++ cmake-2.8.11.2-msys/CMakeCPack.cmake	2014-04-06 22:14:23.853800000 +0400
-@@ -56,6 +56,8 @@
+diff -aur cmake-3.5.2/CMakeCPack.cmake.orig cmake-3.5.2/CMakeCPack.cmake
+--- cmake-3.5.2/CMakeCPack.cmake.orig	2016-04-15 09:41:19.000000000 -0400
++++ cmake-3.5.2/CMakeCPack.cmake	2016-05-22 17:29:34.072190400 -0400
+@@ -54,6 +54,8 @@
      # cygwin is good for the system name
-     if("${CMAKE_SYSTEM_NAME}" STREQUAL "CYGWIN")
+     if("x${CMAKE_SYSTEM_NAME}" STREQUAL "xCYGWIN")
```
Comment (mati865) [inline cmake/cmake-3.5.2-msys.patch]: Wouldn't this `if("x${CMAKE_SYSTEM_NAME}" STREQUAL "xCYGWIN")` break Cmake for CYGWIN?

**Comment by JPeterMugaas** [comment]
I don't know although it might be concerning.  All I can say is that it is what I had found from the CMake source-code distribution but I also doubt how consequential it is considering that the line is only a reference in a patch.    I appreciate the feedback please keep it coming.

BTW, just as an aside, here is how I had found what I found when doing the patch you commented on earlier.  It's just a simple little dirty trick:

find cmake-3.5.2 -name "*.cxx" -exec grep -nH "CYGWIN" {} \;

**Comment by mati865** [comment]
I've used Windows Grep tool to search for CYGWIN but looks like there was a problem with it.

**Comment by JPeterMugaas** [comment]
I have for you, mati865, did I omit anything in this PR from your previous work on cmake?

**Comment by mati865** [comment]
By a quick look I didn't see anything missing. However since deadline my university project changed I'm out for few days.

**Pull request merged**